Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Overrides/PrefabOverrideHandler.cpp

```diff
@@ -62,10 +62,62 @@
             return false;
         }
 
-        AZStd::optional<OverrideType> PrefabOverrideHandler::GetOverrideType(AZ::Dom::Path path, LinkId linkId) const
+        AZStd::optional<OverrideType> PrefabOverrideHandler::GetEntityOverrideType(AZ::Dom::Path path, LinkId linkId) const
         {
             AZStd::optional<OverrideType> overrideType = {};
 
+            AZStd::optional<PatchType> patchType = GetPatchType(path, linkId);
+            if (patchType.has_value())
+            {
+                switch (patchType.value())
+                {
+                case PatchType::Add:
+                    overrideType = OverrideType::AddEntity;
+                    break;
+                case PatchType::Remove:
+                    overrideType = OverrideType::RemoveEntity;
+                    break;
+                case PatchType::Edit:
+                    overrideType = OverrideType::EditEntity;
+                    break;
+                default:
+                    break;
+                }
+            }
+
+            return overrideType;
+        }
+
+        AZStd::optional<OverrideType> PrefabOverrideHandler::GetComponentOverrideType(AZ::Dom::Path path, LinkId linkId) const
```

Comment:
I don't think this class should have 2 methods for fetching entity and component override types. Irrespective of it being an entity or a component, the logic to fetch the type is the same. So, we can just have a single method called GetOverrideType. And the PrefabPublicHandler can have 2 different methods that convert these types to component/entity types as needed.

Reply:
The reason for the two methods is to distinguish between the user wanting the override type on entity scope or on a component scope.

Reply:
Yes. I'm basically suggesting that we move this calculation logic under PrefabOverridePublicHandler::GetComponentOverrideType itself because that function still accepts a component ref as a variable. 

So, for a given api (PrefabOverridePublicHandler), saying 'I'll give you the component, you give me the component override type' makes more sense than, doing something for PrefabOverrideHandler like 'I'll give you the path, you give me the component override type'.  Because what if a path corresponding to a 'remove-entity' patch is provided to this function? In that case it would return 'remove-component' as the type. So the api is ambiguous.

Reply:
Makes sense. Changed.